### PR TITLE
Added get_used_cells_by_id method 3.0

### PR DESCRIPTION
**Pull request by Hassan-A**
same as #9975 but for 3.0

### CODE REVIEW

**Comment by akien-mga** [comment]
Note that you don't need to write "3.0" in the commit log, the fact that it's in the master branch already makes it obvious what version it's included it :)

**Comment by Hassan-A** [comment]
ok great :)
by the way do you think "get used cells by name" would be useful?

**Comment by akien-mga** [comment]
> by the way do you think "get used cells by name" would be useful?

It might be in some cases, but I guess not for most users. Godot's policy is to implement stuff only if it answer a real use case that many users require, so it's probably best not to implement it for now, and see if some users request it.

**Comment by akien-mga** [comment]
Especially since there's already `TileSet::find_tile_by_name(name)` which returns the tile ID for a given name, so this method + the one you just implemented cover this use case.

**Comment by akien-mga** [comment]
(Also I noticed there's a documentation bug in https://github.com/Hassan-A/godot/blob/2255d92e2b53dae9489465c712546f831c2dfe22/scene/resources/tile_set.cpp#L553, that method does not take any argument - but that's not related to your PR of course :P)

**Comment by Hassan-A** [comment]
Sorry, this is my first time how long does it take to merge?

File: scene/2d/tile_map.cpp [hunk 1]
```diff
@@ -1144,6 +1144,24 @@ Array TileMap::get_used_cells() const {
...
+
+			if (E->value().id == p_id) {
+			Vector2 p(E->key().x, E->key().y);
+			a.push_back(p);
+			
+		}
+		
+	}
+	
+		return a;
```
Comment (akien-mga) [inline scene/2d/tile_map.cpp]: Indentation issue here.

Also for the style, no need to keep empty lines before `}`.

Comment (akien-mga) [inline scene/2d/tile_map.cpp]: The style you used in #9975 was good.

File: scene/2d/tile_map.cpp [hunk 2]
```diff
@@ -1144,6 +1144,21 @@ Array TileMap::get_used_cells() const {
 	return a;
 }
 
+Array TileMap::get_used_cells_by_id(int p_id) const {
+
+	Array a;
+	a.clear();
+	for (Map<PosKey, Cell>::Element *E = tile_map.front(); E; E = E->next()) {
+
+			if (E->value().id == p_id) {
```
Comment (akien-mga) [inline scene/2d/tile_map.cpp]: Sadly there's still an indentation issue here, it should be one tab less.

File: scene/2d/tile_map.cpp [hunk 3]
```diff
@@ -1144,6 +1144,21 @@ Array TileMap::get_used_cells() const {
 	return a;
 }
 
+Array TileMap::get_used_cells_by_id(int p_id) const {
+
+	Array a;
+	a.clear();
```
Comment (akien-mga) [inline scene/2d/tile_map.cpp]: I guess the `a.clear()` is not necessary here, `Array a;` already declares it empty.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
All good, thanks for your patience. I'll merge once Travis CI is happy.

**Comment by Hassan-A** [comment]
Great thanks for your help.

**Comment by akien-mga** [comment]
If you're interested, we have some documentation on the code style, and especially about the `clang-format` tool we use to enforce it. You can install it to run it locally on your changes and verify that they are good: http://docs.godotengine.org/en/latest/community/contributing/code_style_guidelines.html

File: scene/2d/tile_map.cpp [hunk 4]
```diff
@@ -1144,6 +1144,20 @@ Array TileMap::get_used_cells() const {
...
+
+	Array a;
+	for (Map<PosKey, Cell>::Element *E = tile_map.front(); E; E = E->next()) {
+
+		if (E->value().id == p_id) {
+			Vector2 p(E->key().x, E->key().y);
+			a.push_back(p);
+		}
+	}
+
```
Comment (bojidar-bg) [inline scene/2d/tile_map.cpp]: Trailing whitespace here.

**Comment by 2fd5** [comment]
@Hassan-A @akien-mga  will this be added to 4.0 ? as I cannot see it there

**Comment by akien-mga** [comment]
Please open a dedicated issue. Commenting on merged commits or 5 year old PRs is not the best workflow to get issues discussed and resolved.

**Pull request merged**